common/math/spline_interpolation/CMakeLists.txt

```diff
@@ -1,43 +1,14 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.5)
 project(spline_interpolation)
 
-add_compile_options(-std=c++14)
+if(NOT CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 14)
+endif()
```

Comment:
There's some more boilerplate we usually add:
```
if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
  add_compile_options(-Wno-unused-parameter -Wall -Wextra -Wpedantic)
endif()
```
I think Fred has even more in his newest porting doc PR

Reply:
@nnmm yes, that's true but we can make the standard enforcement stricter in a 2nd step.

@cvasfi please just add the warning flags, and while building temporarily add `-Werror` to make sure there are no  warnings left but don't commit `-Werror`

Reply:
Looks like `Werror` doesn't show any error so I will merge this.